cmd/openshift-install/create.go

```diff
@@ -186,6 +193,49 @@ func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args
 	}
 }
 
+// appendRouterCAToClusterCA appends router CA to cluster CA in kubeconfig
+func appendRouterCAToClusterCA(config *rest.Config, directory string) (err error) {
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, "creating a Kubernetes client")
+	}
+
+	routerSecret, err := client.CoreV1().Secrets("openshift-ingress-operator").Get("router-ca", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	routerCrtBytes := routerSecret.Data["tls.crt"]
+	absDir, err := filepath.Abs(directory)
+	if err != nil {
+		return err
+	}
+	kubeconfig := filepath.Join(absDir, "auth", "kubeconfig")
+	yml, err := ioutil.ReadFile(kubeconfig)
+	if err != nil {
+		return err
+	}
+	var kconfig *clientcmdv1.Config
+	err = yaml.Unmarshal(yml, &kconfig)
+	if err != nil {
+		return err
+	}
+
+	clusterCABytes := kconfig.Clusters[0].Cluster.CertificateAuthorityData
+	appendedCA := string(routerCrtBytes) + string(clusterCABytes)
```

Comment:
"Appended" is not quite the right word here since you are adding the router CA to the beginning not the end.

Reply:
correct, thanks, i'll rename : )

Reply:
You can use [clientcmd.LoadFromFile](https://godoc.org/k8s.io/client-go/tools/clientcmd#LoadFromFile) and [clientcmd.WriteToFile](https://godoc.org/k8s.io/client-go/tools/clientcmd#WriteToFile`) . Might want to use certpool helpers to concat the CAs as well

Reply:
ah, awesome, thanks